cli/cmd/upgrade/upgrade.go

```diff
@@ -0,0 +1,453 @@
+package upgrade
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	consulChart "github.com/hashicorp/consul-k8s/charts"
+	"helm.sh/helm/v3/pkg/chart/loader"
+
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/cli/values"
+	"helm.sh/helm/v3/pkg/getter"
+
+	"github.com/hashicorp/consul-k8s/cli/cmd/common/terminal"
+	"github.com/hashicorp/consul-k8s/cli/config"
+	helmCLI "helm.sh/helm/v3/pkg/cli"
+
+	"github.com/hashicorp/consul-k8s/cli/cmd/install"
+
+	"github.com/hashicorp/consul-k8s/cli/cmd/common"
+	"github.com/hashicorp/consul-k8s/cli/cmd/common/flag"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	flagNamePreset = "preset"
+	defaultPreset  = ""
+
+	flagNameConfigFile      = "config-file"
+	flagNameSetStringValues = "set-string"
+	flagNameSetValues       = "set"
+	flagNameFileValues      = "set-file"
+
+	flagNameDryRun = "dry-run"
+	defaultDryRun  = false
+
+	flagNameAutoApprove = "auto-approve"
+	defaultAutoApprove  = false
+
+	flagNamespace    = "namespace"
+	defaultNamespace = "consul"
+
+	flagNameTimeout = "timeout"
+	defaultTimeout  = "10m"
+
+	flagNameVerbose = "verbose"
+	defaultVerbose  = false
+
+	flagNameWait = "wait"
+	defaultWait  = true
+)
+
+type Command struct {
+	*common.BaseCommand
+
+	kubernetes kubernetes.Interface
+
+	set *flag.Sets
+
+	flagPreset          string
+	flagDryRun          bool
+	flagAutoApprove     bool
+	flagNamespace       string
+	flagValueFiles      []string
+	flagSetStringValues []string
+	flagSetValues       []string
+	flagFileValues      []string
+	flagTimeout         string
+	timeoutDuration     time.Duration
+	flagVerbose         bool
+	flagWait            bool
+
+	flagKubeConfig  string
+	flagKubeContext string
+
+	once sync.Once
+	help string
+}
+
+func (c *Command) init() {
+	// Store all the possible preset values in 'presetList'. Printed in the help message.
+	var presetList []string
+	for name := range config.Presets {
+		presetList = append(presetList, name)
+	}
+
+	c.set = flag.NewSets()
+	f := c.set.NewSet("Command Options")
+	f.BoolVar(&flag.BoolVar{
+		Name:    flagNameAutoApprove,
+		Target:  &c.flagAutoApprove,
+		Default: defaultAutoApprove,
+		Usage:   "Skip confirmation prompt.",
+	})
+	f.StringVar(&flag.StringVar{
```

Comment:
I don't think we're using the namespace flag anywhere, since we should just use whatever namespace the existing installation is in. This and L45-46, L68 can be removed

Reply:
And the check inside of validateFlags function (L378) can be removed